extensions/amp-viewer-integration/0.1/messaging/messaging.js

```diff
@@ -83,13 +83,9 @@ export class WindowPortEmulator {
    * @param {function(!Event):undefined} handler
    */
   addEventListener(eventType, handler) {
-    this.win.addEventListener('message', e => {
-      if (
-        e.origin == this.origin_ &&
-        e.source == this.target_ &&
-        e.data['app'] === APP
```

Comment:
FYI: I moved `e.data['app'] === APP` to `Messaging#handleMessage_` - it makes more sense to check there, I think, since it covers `MessagePort` as well.